apps/ex_wire/lib/ex_wire/struct/bucket.ex

```diff
@@ -0,0 +1,273 @@
+defmodule ExWire.Struct.Bucket do
+  @moduledoc """
+  Represents a Kademlia k-bucket.
+  """
+
+  alias ExWire.Struct.{Peer, Bucket}
+  alias ExWire.Util.Timestamp
+  alias ExWire.KademliaConfig
+
+  defstruct [:id, :nodes, :updated_at]
+
+  @type t :: %__MODULE__{
+          id: integer(),
+          nodes: [Peer.t()],
+          updated_at: integer()
+        }
+
+  @doc """
+  Creates new bucket.
+
+  ## Examples
+      iex> ExWire.Struct.Bucket.new(1, time: :test)
+      %ExWire.Struct.Bucket{
+        id: 1,
+        nodes: [],
+        updated_at: 1525704921
+      }
+
+  """
+  @spec new(Peer.t()) :: t()
+  def new(id, options \\ [time: :actual]) do
+    %__MODULE__{
+      id: id,
+      nodes: [],
```

Comment:
Maybe list is ok for the initial implementation, but we need order set (no such thing built in elixir yet), I'll try to checkout the branch and replace the list with [rbtree](https://hexdocs.pm/rbtree/Tree.html) implementation

Reply:
Ok, just tried to update your code to used it instead of list, basically it works, but there are too many changes needed to use rbtree which out of scope of this PR, also I don't like [how rbtree is written](https://github.com/vyorkin-forks/rbtree/blob/master/lib/rbtree.ex#L519). Lets just use lists for now and create a separate issue to address this later